anagram.cpp

```diff
@@ -1,13 +1,61 @@
 #include "anagram.h"
+#include<algorithm>
+using namespace std;
 
-bool Anagram::WordPairIsAnagram(const std::string& word1, const std::string& word2) {
-    //Fill the correct implementation here
+
+string arrange(string& str)
+{
+    int count = 0;string str2;
+    for (unsigned int i = 0;i < str.length();i++)
+    {
+        if (str[i] != ' ')
+        {
+            str[count] = str[i];
+            count += 1;
+        }
+       str2 = str.substr(0, count);
+    }transform(str2.begin(), str2.end(), str2.begin(), ::tolower);
+    sort(str2.begin(), str2.end());
+    return str2;
+}
+
+bool checklength( string& word1,  string& word2)
+{
+    if (word1.length() == word2.length())
+    {   return true;    }     
     return false;
 }
+bool checkChar( string& word1,  string& word2)
+{
+    for (unsigned int i = 0;i < word1.length();i++)
+    {
+        if (word1[i] != word2[i])
+        {
+            return  false;
+        }
+    }
+    return true;
+}
+bool Anagram::WordPairIsAnagram(const string& word1, const string& word2) {
+    string word1copy = word1;string word2copy = word2;
```

Comment:
Add tests for this function - see examples in Wikipedia anagrams, especially the first three.